### DISCUSSION TITLE
WinSetView command line / batch mode

### DISCUSSION THREAD

**Discussion by gaiking-uk** [LesFerch/WinSetView#112] [Q&A] [answered]
### TLDR;
- Is it possible to call **WinSetView** via the command line, passing in parameters such as the folder path and desired view type?

### Context
- I have a set of media folders with a fairly standard structure - each folder has an "Images" sub-folder, a "Videos" sub-folder, etc
- Currently every time I create a new project, I have to manually set the view type on the "Videos" folder to videos, and the view type on the "Images" folder to "Pictures", etc 😴💤
- It would be great if I could make repeated **WinSetView** calls, along the lines of... `WinSetView.exe -Path "D:\Media projects\Project 1\Videos" -Type "Videos"` etc

### Example Structure / Requirements
```
D:\Media Projects
    └ Project 1
       └ Assets
          └ Audio     <-- Set folder type as Music
          └ Images    <-- Set folder type as Pictures
          └ Video     <-- Set folder type as Videos
    └ Project 2
       └ Assets
          └ Audio     <-- Set folder type as Music
          └ Images    <-- Set folder type as Pictures
          └ Video     <-- Set folder type as Videos
    └ Project 3
       └ Assets
          └ Audio     <-- Set folder type as Music
          └ Images    <-- Set folder type as Pictures
          └ Video     <-- Set folder type as Videos
    └ Project 4
       └ Assets
          └ Audio     ... etc ...
```

### Is this possible?
- Hopefully yes! 🙏🏼
- If not, is this something that could be added?... or is there a different tool / way of doing this?

**Comment by LesFerch** [comment, marked as answer]
WinSetView does not actually deal with individual folders. It just sets your preferred defaults in the registry for each folder type and then clears all saved views. Explorer applies the view when you first open a folder.

For the folder type, Explorer automatically sets the type based on the majority content. Windows 11 is very aggressive about doing that. So much so that you cannot override what it picks. Windows 10 sets the folder type based on content until you manually set it and then it backs off (which is the way it should work). So, in theory, you shouldn't have to set the folder type. It should set automatically when you add content, unless you have enabled the option "Make all folders Generic".

Since Microsoft provides no API for setting the folder type, the only option to script it is to set the folder type with a hidden **desktop.ini** file. I think I could write you a script that will traverse your folder structure looking for folders named **Audio**, **Images**, and **Video** and set the folder types using hidden deskop.ini files. This method will only work on NTFS volumes that mount as "Local Disk" (like C: drive), so hopefully that's what we're working with.
